test/integration_tests.jl

```diff
@@ -0,0 +1,163 @@
+
+# ramsey evolution of a single transmon should produce sinusoidal oscillations at the qubit frequency
+@testset "Ramsey" begin
+    dims = 3
+    qubit_freq = 5.0
+    anharm = -0.2
+    q0 = DuffingTransmon("q0", dims, DuffingSpec(qubit_freq, anharm))
+    cqs = CompositeQSystem([q0])
+    add_hamiltonian!(cqs, q0)
+    times = collect(range(0,stop=1,length=201))
+    ψ_init = (1/sqrt(2)) * ComplexF64[1; 1; 0]
+    ψs = unitary_state(cqs, times, ψ_init)
+    signal = Float64[real(ψ_init'*ψ) for ψ in ψs]
+    expected = 0.5 .+ 0.5*cos.(2π*qubit_freq * times)
+    @test isapprox(signal, expected, rtol=1e-3)
+
+    # check that the propagator gives the same result
+    us = unitary_propagator(cqs, times)
+    ψs = [u*ψ_init for u in us]
+    signal = Float64[real(ψ_init'*ψ) for ψ in ψs]
+    @test isapprox(signal, expected, rtol=1e-4)
+end
+
+########################### Rabi ##########################
+
+# rabi flops should produce sinusoidal oscillations between ground and excited state
+@testset "Rabi" begin
+    dims = 3
+    qubit_freq = 5.0
+    anharm = -0.2
+    nutation_freq = 0.02
+    q0 = DuffingTransmon("q0", dims, DuffingSpec(qubit_freq, anharm))
+    cqs = CompositeQSystem([q0])
+    add_hamiltonian!(cqs, q0)
+    add_hamiltonian!(cqs, dipole_drive(q0, t -> nutation_freq*cos(2π*qubit_freq * t)), q0)
+    ψ_init = ComplexF64[1; 0; 0]
+    times = collect(range(0,stop=100,length=101))
+    ψs = unitary_state(cqs, times, ψ_init)
+
+    g_sim = [abs2(s[1]) for s in ψs]
+    e_sim = [abs2(s[2]) for s in ψs]
+
+    # we would normally expect a factor of 0.5 from lab -> rotating frame but by defining the qubit
+    # drive in terms of X rather than 0.5 X undoes that
+    g_expected = 0.5 .+ 0.5*cos.(2π*nutation_freq * times)
+    @test isapprox(g_sim, g_expected, rtol=1e-2)
+    @test isapprox(e_sim, 1 .- g_expected, rtol=1e-2)
+
+    #rwa version
+    cqs_rwa = CompositeQSystem([q0])
+    add_hamiltonian!(cqs_rwa, q0)
+    add_hamiltonian!(cqs_rwa, rwa_dipole(q0, t -> nutation_freq*cos(2π*qubit_freq * t)), q0)
+    ψs_rwa = unitary_state(cqs_rwa, times, ψ_init)
+
+    g_sim_rwa = [abs2(s[1]) for s in ψs_rwa]
+    e_sim_rwa = [abs2(s[2]) for s in ψs_rwa]
+
+    @test isapprox(g_sim_rwa, g_expected, rtol=1e-2)
+    @test isapprox(e_sim_rwa, 1 .- g_expected, rtol=1e-2)
+
+    # check that the propagator gives the same result
+    # use floquet_propagator to speed it up since the Hamiltonian is periodic
+    floquet_prop = floquet_propagator(unitary_propagator, 1/qubit_freq)
+    us = floquet_prop(cqs, times)
+    ψs = [u*ψ_init for u in us]
+    g_sim = [abs2(s[1]) for s in ψs]
+    e_sim = [abs2(s[2]) for s in ψs]
+    @test isapprox(g_sim, g_expected; rtol=1e-2)
+    @test isapprox(e_sim, 1 .- g_expected; rtol=1e-2)
+end
+
+########################### Parmetric Flux Drive ##########################
+
+# parametric flux drive should produce flops between resonant states
+# parameters from Blue Launch paper
```

Comment:
Can you be more specific than "Blue Launch paper". e.g. "S. Caldwell et al,  Phys. Rev. Applied 10, 034050 (2018)"